src/integrations/prefect-redis/prefect_redis/lease_storage.py

```diff
@@ -166,12 +251,37 @@ async def renew_lease(self, lease_id: UUID, ttl: timedelta) -> None:
     async def revoke_lease(self, lease_id: UUID) -> None:
         try:
             lease_key = self._lease_key(lease_id)
+            # Read lease to clean up indexes
+            data = await self.redis_client.get(lease_key)
+            holder_key: str | None = None
+            resource_ids: list[UUID] = []
+            if data is not None:
+                lease = self._deserialize_lease(data)
+                resource_ids = lease.resource_ids
+                holder = (
+                    getattr(lease.metadata, "holder", None) if lease.metadata else None
+                )
+                if hasattr(holder, "model_dump"):
+                    holder = holder.model_dump(mode="json")  # type: ignore[attr-defined]
+                holder_key = self._holder_key(holder)
```

Comment:
Can you include the read operation in the revoke Lua script? There's a possible race condition here if the lease key is removed between read and the revoke Lua script.